selfservice/strategy/saml/strategy_registration.go

```diff
@@ -0,0 +1,138 @@
+package saml
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/ory/kratos/identity"
+	"github.com/ory/x/decoderx"
+
+	"github.com/ory/kratos/selfservice/flow"
+	"github.com/ory/kratos/selfservice/flow/registration"
+	"github.com/ory/kratos/text"
+
+	"github.com/tidwall/sjson"
+
+	"github.com/ory/kratos/x"
+)
+
+// Implement the interface
+var _ registration.Strategy = new(Strategy)
+
+// Call at the creation of Kratos, when Kratos implement all authentication routes
+func (s *Strategy) RegisterRegistrationRoutes(r *x.RouterPublic) {
+	s.setRoutes(r)
+}
+
+func (s *Strategy) createIdentity(w http.ResponseWriter, r *http.Request, a *registration.Flow, claims *Claims, provider Provider) (*identity.Identity, error) {
+	var jsonClaims bytes.Buffer
+	if err := json.NewEncoder(&jsonClaims).Encode(claims); err != nil {
+		return nil, s.handleError(w, r, a, provider.Config().ID, nil, err)
+	}
+
+	i := identity.NewIdentity(s.d.Config().DefaultIdentityTraitsSchemaID(r.Context()))
+	if err := s.setTraits(w, r, claims, provider, jsonClaims, i); err != nil {
```

Comment:
Naive approach to traits mapping, will not work in practice. Need to add JsonNet parsing and more